lti_consumer/api.py

```diff
@@ -180,29 +190,65 @@ def get_lti_1p3_content_url(config_id=None, block=None, hint=""):
         b. Deep Linking is configured, but a single ltiResourceLink was selected.
     2. The Deep Linking content presentation URL if there's more than one
        Lti DL content in the database.
+
+    The launch_data parameter is a dictionary that contains data necessary and related to an LTI 1.3 launch.
+    It provides a mechanism for the context to provide contextually defined or stored launch information to the library.
+    The schema of launch_data may change as we support additional claims or LTI specifications.
+
+    Below are the currently support keys for launch_data.
+
+    config_id (required; string): the config_id field of an LtiConfiguration to use for the launch
+    user_role (required; string): the user's role as one of the keys in LTI_1P3_ROLE_MAP: staff, instructor, student,
+        or guest.
+    user_id (required; string, int): the user's unique identifier
+    resource_link_id (required; string): a unique identifier that is guaranteed to be unique within the platform for
```

Comment:
@MichaelRoytman Just a note here: `resource_link_id` should be unique **per placement**, not per platform. This means that each place that has an LTI 1.3 integration needs to have a unique ID - in the previous implementation this was easily solvable by using the XBlock ID, but I'm not sure how are we going to do this now. My guess is that for XBlock launches, we should keep using the XBlock ID (it also makes linking grades easier). What do you think?

Reply:
Thank you for the clarification! I'll update the comment.

Are you comfortable with allowing the context to supply the `resource_link_id`? I ask because you mentioned having an `LtiPlacement` model in your [feedback](https://github.com/openedx/xblock-lti-consumer/pull/281/files#r978036669) in the draft ADR. I assume that the library would get the `resource_link_id` from the `LtiPlacement` model instead of getting it as a part of `launch_data`.